addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRule.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2020 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpStatusCode;
+
+/**
+ * Active scan rule for Reverse proxy testing.
+ *
+ * @author Aman Rawat (@theamanrawat)
+ */
+public class ForbiddenBypassScanRule extends AbstractAppPlugin {
+
+    private static Logger log = Logger.getLogger(ForbiddenBypassScanRule.class);
+
+    @Override
+    public int getId() {
+        return 40038;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString("ascanalpha.forbiddenBypass.name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString("ascanalpha.forbiddenBypass.desc");
+    }
+
+    @Override
+    public void scan() {
+        HttpMessage message = getNewMsg();
+        try {
+            sendAndReceive(message);
+            if (message.getResponseHeader().getStatusCode() == HttpStatusCode.FORBIDDEN) {
```

Comment:
Consider returning earlier instead of nesting.

Reply:
I didn't get this can you please tell more about what should I return earlier.

Reply:
Just return, e.g.:
```Java
if (message.getResponseHeader().getStatusCode() != HttpStatusCode.FORBIDDEN) {
    return;
}
```

Reply:
This was for the existing if.

Reply:
Where should I return in this code like this one

```java
 if (message.getResponseHeader().getStatusCode() == HttpStatusCode.FORBIDDEN) {
        URI uri = message.getRequestHeader().getURI();
        String host = uri.getEscapedAuthority();
        String path = uri.getEscapedPath();
        String schema = uri.getScheme();
        if(sendPathPayloads(path, host, schema, uri)){
              return;
         };
        if(sendHeaderPayloads(path, host, schema, uri)){
               return;
         };

        return; //this one 
 }
```

Reply:
```java
if (message.getResponseHeader().getStatusCode() != HttpStatusCode.FORBIDDEN) {
    return;
}

URI uri = message.getRequestHeader().getURI();
String host = uri.getEscapedAuthority();
String path = uri.getEscapedPath();
String schema = uri.getScheme();
if (sendPathPayloads(path, host, schema, uri)) {
    return;
}
if (sendHeaderPayloads(path, host, schema, uri)) {
    return;
}
```

Reply:
Thanks.